receiver/snowflakereceiver/scraper.go

```diff
@@ -48,34 +49,76 @@ func (s *snowflakeMetricsScraper) shutdown(_ context.Context) (err error) {
 	return err
 }
 
+func errorListener(ctx context.Context, eQueue <-chan error, eOut chan<- *scrapererror.ScrapeErrors) {
+	errs := &scrapererror.ScrapeErrors{}
+
+	for {
+		select {
+		case <-ctx.Done():
+			eOut <- errs
+			return
+		case err, ok := <-eQueue:
+			if err == nil || !ok {
```

Comment:
You don't need to send nil to the channel, just close the `eQueue` instead. 
```suggestion
			if !ok {
```